### PR TITLE
Fix build issues

### LINKED ISSUE #667: Travis CI build fails
### Overview Description
The travis CI build for corobo is failing.
# Steps to Reproduce

1. Go to https://travis-ci.com
2. Run travis CI build for the coala/corobo repository

# Actual Results
Following are the errors occuring:
1.
```
tests/git_stats_test.py:4: in <module>
    from IGitt.GitHub.GitHubMergeRequest import GitHubMergeRequest
/usr/local/lib/python3.9/site-packages/IGitt/GitHub/__init__.py:13: in <module>
    from IGitt.Interfaces import Token, get, post
/usr/local/lib/python3.9/site-packages/IGitt/Interfaces/__init__.py:157: in <module>
    def get_response(method: Callable,
/usr/local/lib/python3.9/site-packages/backoff/_decorator.py:153: in decorate
    if asyncio.Task.current_task() is not None:
    AttributeError: type object '_asyncio.Task' has no attribute 'current_task'
```
2.
```
$ python -m pytest
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/pluggy/manager.py", line 253, in load_setuptools_entrypoints
    plugin = ep.load()
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2460, in load
    self.require(*args, **kwargs)
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2483, in require
    items = working_set.resolve(reqs, env, installer, extras=self.extras)
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.VersionConflict: (pytest 3.6.4 (/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages), Requirement.parse('pytest>=4.6'))
```
3.
```
$ python -m pytest
ERROR: usage: __main__.py [options] [file_or_dir] [file_or_dir] [...]
__main__.py: error: unrecognized arguments: --doctest-ignore-import-error
```
4.
```
self = <tests.wolfram_alpha_test.WolframAlphaTest testMethod=test_wa>
    @my_vcr.use_cassette('tests/cassettes/wa.yaml')
    def test_wa(self):
        self.push_message('!wa 2^6')
        self.assertIn('64', self.pop_message())
        with self.assertLogs() as cm:
            self.assertCommand('!wa this is a sentence',
                               'Dunno')
        self.assertIn('INFO:errbot.plugins.WolframAlpha:KeyError triggered on '
>                     'retrieving pods.', cm.output)
       AssertionError: 'INFO:errbot.plugins.WolframAlpha:KeyError triggered on retrieving pods.' not found in
['INFO:errbot.plugins.ACLs:Matching ACL {} against username gbin@localhost for command WolframAlpha:wa.',
'INFO:errbot.plugins.ACLs:Check if wa is admin only command.', 'INFO:errbot.core:Processing command "wa" with
parameters "this is a sentence" from gbin@localhost', 'ERROR:errbot.core:callback_message on deprecate_bot_prefixes
crashed.\nTraceback (most recent call last):\n  File "/usr/local/lib/python3.6/site-packages/errbot/core.py", line 111, in
_dispatch_to_plugins\n    getattr(plugin, method)(*args, **kwargs)\n  File "plugins/deprecate_bot_prefixes.py", line 15, in
callback_message\n    for deprecated_prefix in self.bot_config.BOT_DEPRECATED_PREFIXES:\nAttributeError:
\'ShallowConfig\' object has no attribute \'BOT_DEPRECATED_PREFIXES\'', 'INFO:vcr.stubs:Playing response for <Request
(GET) https://api.wolframalpha.com/v2/query?input=this+is+a+sentence&appid=None> from cassette',
'INFO:errbot.backends.test:\n\n\nMESSAGE:\nDunno :(\n\n\n']
tests/wolfram_alpha_test.py:22: AssertionError
```

# Expected Results
No errors. All tests should pass with 100% test coverage. All commands should exit with 0.

# Reproducibility
Every time

### PR DESCRIPTION
**Pull request by codehobbyist06**
Fixes https://github.com/coala/corobo/issues/667

# Reviewers Checklist

- [ ] Appropriate logging is done.
- [ ] Appropriate error responses.
- [ ] Handle every possible exception.
- [ ] Make sure there is a docstring in the command functions. Hint: Lookout for
  `botcmd` and `re_botcmd` decorators.
- [ ] See that 100% coverage is there.
- [ ] See to it that mocking is not done where it is not necessary.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ before_script:
 
 script:
   - coala --non-interactive -V
+  - pip install --upgrade pytest~=4.6
```
Comment (abhishalya) [inline .travis.yml:34]: This should not be here, you should make changes to the requirements.txt file instead

File: setup.cfg [hunk 2]
```diff
@@ -18,12 +18,13 @@ addopts =
...
   --color=yes
   --doctest-glob=*.rst
-  --doctest-ignore-import-error
+  --doctest-ignore-import-errors
   --error-for-skips
   --cov
   --instafail
   --profile
   --reorder 'requirements.txt' 'test-requirements.txt' '*'
+  --ignore=tests/wolfram_alpha_test.py
```
Comment (abhishalya) [inline setup.cfg:27]: Why is this required?

File: setup.cfg [hunk 3]
```diff
@@ -66,6 +67,7 @@ omit =
   utils/utils.py
   plugins/ban.py
   plugins/labhub.py
+  plugins/wolfram_alpha.py
```
Comment (abhishalya) [inline setup.cfg:70]: Try to fix the issue instead of ignoring it if possible. We will resort to ignoring it as a last option.

**Review by abhishalya — CHANGES_REQUESTED** [review]
Also please squash the commits into one. For a set of changes belonging to same category/issue we would want a single commit representing it.

File: .travis.yml (hunk 1 above)
Comment (codehobbyist06) [reply .travis.yml:34]: I also wanted to change there but the problem is that coala has requirements of pytest==3.6.4 and python requires pytest==4.6.2 for running tests. So, if I update the requirements.txt file, the command ```coala --non-interactive -V``` fails. One thing that can be done is that we can change the requirements.txt file of coala too to pytest==4.6, but I don't know what impact it would have on coala.

File: setup.cfg (hunk 2 above)
Comment (codehobbyist06) [reply setup.cfg:27]: The wolfram_alpha_test is currently not working correctly, it is giving some errors. Did this as a temporary fix.

File: setup.cfg (hunk 3 above)
Comment (codehobbyist06) [reply setup.cfg:70]: Since, I have ignored the wolfram_alpha_test, it is giving no coverage for this plugin, so just ignored it. Once the wolfram_alpha_test is fixed, both the files will be added to the tests.

File: setup.cfg (hunk 2 above)
Comment (abhishalya) [reply setup.cfg:27]: Ignoring is not a solution. We'll need to fix the tests.

File: .travis.yml (hunk 1 above)
Comment (abhishalya) [reply .travis.yml:34]: Pytest version for this can be changed. coala right now has a strict pytest dependency but here we can adjust.

File: setup.cfg (hunk 3 above)
Comment (abhishalya) [reply setup.cfg:70]: The fix should go before we merge this then.

Comment (codehobbyist06) [reply setup.cfg:70]: Sure, will do this.

File: setup.cfg (hunk 2 above)
Comment (codehobbyist06) [reply setup.cfg:27]: Yes, I know. I am working on that, will get this done as early as possible.

File: setup.cfg (hunk 3 above)
Comment (codehobbyist06) [reply setup.cfg:70]: I will get this one done and then squash all the commits.

File: .travis.yml (hunk 1 above)
Comment (codehobbyist06) [reply .travis.yml:34]: Thats what I am trying to ask, that if we change the pytest version in corobo then the coala check that we do in travis CI builds would fail(I have tried this out). So updating pytest before running the tests only seemed me to be a solution that would compromise this requirement conflict.

Comment (abhishalya) [reply .travis.yml:34]: coala doesn't need pytest to run. It is used to run tests only. pytest is a part of coala's test-requirements and not its requirements. So the dependency surely doesn't come from coala. The failure you see is because of something else.

Comment (codehobbyist06) [reply .travis.yml:34]: I also thought that but the error clearly says that there coala requirements are not satisfied and the traceback also passes through the coala_main.py file. Logically, it doesn't seem correct that test-requirements should be satisfied for coala to run. But it seems like this. I will investigate this. By then, Can you pls have a look at these logs and suggest your views:
```
$ coala --non-interactive -V
[WARNING][06:28:39] bears setting has already been defined in section all.python. The previous setting will be overridden.
[WARNING][06:28:39] bears setting has already been defined in section all.python. The previous setting will be overridden.
[ERROR][06:28:39] There is a conflict in the version of a dependency you have installed and the requirements of coala. This may be resolved by creating a separate virtual environment for coala or running `pip3 install "importlib-metadata>=1.7.0; python_version < "3.8""`. Be aware that the latter solution might break other python packages that depend on the currently installed version.
[INFO][06:28:39] Exception was:
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/coalib/coala_main.py", line 147, in run_coala
    args=args)
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/coalib/settings/ConfigurationGathering.py", line 504, in gather_configuration
    targets=targets,
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/coalib/settings/SectionFilling.py", line 97, in fill_settings
    bear_dirs = section.bear_dirs()
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/coalib/settings/Section.py", line 147, in bear_dirs
    for bear_dir in collect_registered_bears_dirs('coalabears')]
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/coalib/collecting/Collectors.py", line 456, in collect_registered_bears_dirs
    registered_package = ep.load()
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2460, in load
    self.require(*args, **kwargs)
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2483, in require
    items = working_set.resolve(reqs, env, installer, extras=self.extras)
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (importlib-metadata 0.18 (/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages), Requirement.parse('importlib-metadata>=1.7.0; python_version < "3.8"'), {'stevedore'})
The command "coala --non-interactive -V" exited with 13.
```

File: setup.cfg (hunk 3 above)
Comment (codehobbyist06) [reply setup.cfg]: Done with the changes. The builds are working and no file is ignored. Also, no upgrade command in travis.yml in script section.

**Review by abhishalya — APPROVED** [review]
Nice work! Although, since we're unable to trigger any travis builds here, can you link a build from your fork here?

**Comment by codehobbyist06** [comment]
Sure. Following is the travis CI build:
https://travis-ci.com/github/codehobbyist06/corobo/builds/223374656
